### DISCUSSION TITLE
changing window height

### DISCUSSION THREAD

**Discussion by d3m1gd** [YaLTeR/niri#474] [Ideas]
mouse window resize and keybind window resize allow resizing window down beyond the bottoms edge of the screen. maybe the feature is intended, but it breaks the apparent rigidity of the window stripe, with overall experience it is being strictly bounded.

i propose a toggle to restrict resizing beyond bottom and top when resizing a window.

going further, it would be awesome to have a toggle to prevent vertical resize if there is a single window in a column, so that outer column edges are fixed, and only the split can be moved up and down.

**Comment by YaLTeR** [comment]
It is intended, yeah. Higher than screen height for taking screenshots of specific window sizes, and smaller than screen height is generally useful for testing and for preserving fixed height.

> breaks the apparent rigidity of the window stripe

Well, there are plenty of cases when it will break anyway. Windows will be smaller if they want, e.g. dialogs, mpv with CSD with its aspect ratio enforcement, terminals which resize in grid increments. Windows will also be bigger if they want, e.g. windows that don't fit in the provided height (easiest to trigger by stacking several windows in a column), or again mpv with CSD with aspect ratio enforcement.

All in all this kind of turns into a complicated code chase with a thousand edge cases.

**Reply by d3m1gd** [reply]
while all those cases are certainly true, there is somewhat bad user experience, when i try to resize column with a mouse horizontally and inevitably it wil shift up or down be at least several pixels, and become uneven, so another key press is required to reset it to the original height

edit: 

so, maybe expose gestures as binds, and one of gestures could be horizontal-resize, another vertical-resize and  a simple resize, which is all directional

**Reply by YaLTeR** [reply]
Fwiw if you start a resize closer to the vertical center, it will be horizontal-only.

**Comment by YaLTeR** [comment]
I made a new discussion where I wrote everything down, please suggest things there: https://github.com/YaLTeR/niri/discussions/593
